pkg/cli/admin/release/new.go

```diff
@@ -807,11 +807,15 @@ func resolveImageStreamTagsToReferenceMode(inputIS, is *imageapi.ImageStream, re
 
 		externalFn := func(source, image string) string {
 			// filter source URLs
-			if len(source) > 0 && len(internal) > 0 && strings.HasPrefix(source, internal) {
+			if !toMirror && len(source) > 0 && len(internal) > 0 && strings.HasPrefix(source, internal) {
 				klog.V(2).Infof("Can't use source %s because it points to the internal registry", source)
 				source = ""
 			}
-			// default to the external registry name
+			// if --mirror is set and the referenceMode is not 'source', use the internal registry as image source.
```

Comment:
`mirror` field can only be set for the referenceMode public (see: https://github.com/openshift/oc/blob/1f30e7346ee765cb27d4ac451720e8fd30a01be8/pkg/cli/admin/release/new.go#L284) and public referenceMode is self-descriptive that can force using external. Besides, I'm not sure I fully measure the consequences of this change.  

Yes, it makes sense from the pod as explained but it still (incorrectly) may fall back to internal registry even in the connected environment instead surfacing to external registry.